examples/benchmark/src/main.c

```diff
@@ -0,0 +1,1361 @@
+#include <mpix/image.h>
+#include <mpix/low_level.h>
+#include <mpix/port.h>
+#include <mpix/print.h>
+#include <mpix/types.h>
+#include <mpix/utils.h>
+#include <mpix/stats.h>
+
+#include "benchmark.h"
+#include "helpers.h"
+
+#ifndef CONFIG_WIDTH
+#define CONFIG_WIDTH 16
+#endif
+
+#ifndef CONFIG_HEIGHT
+#define CONFIG_HEIGHT 16
+#endif
+
+#define WIDTH CONFIG_WIDTH
+#define HEIGHT CONFIG_HEIGHT
+#define RGB24_CHANNELS 3
+#define BRG8_CHANNELS 1
+
+#define NUM_OF_BENCHMARKS 16
+
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
+
+#define RES_STR(a, b) STR(a) "x" STR(b)
+
+#define CHECK(x) \
+	({ int e = (x); if (e) { fprintf(stderr, "%s: %s\n", #x, strerror(-e)); return e; } })
+
+int benchmark_debayer_1x1(struct benchmark_entry *entry, uint16_t width,
+                          uint16_t height, uint32_t format)
+{
+  struct mpix_image img = {};
+  struct mpix_format fmt = {.width = width, .height = height, .fourcc = format};
+
+  size_t img_size = mpix_format_pitch(&fmt) * fmt.height;
+  size_t outbuf_size = width * height * 3;
+  uint8_t *srcbuf = mpix_port_alloc(img_size, MPIX_MEM_SOURCE_DEFAULT);
+  uint8_t *outbuf = mpix_port_alloc(outbuf_size, MPIX_MEM_SOURCE_DEFAULT);
```

Comment:
Instead of using malloc/free (which can fail, need to check error status), it's possible to use a global [1] static array that is large enough for every pixel format type (3 byte per pixel should be enough).

[1]: important to make it global so that it does not use stack memory, which is restricted on MCU, unlike global arrays.